src/System.Diagnostics.Tracing.Telemetry/src/System/Diagnostics/Tracing/TelemetrySource.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// This is the basic API to 'hook' parts of the framework.   It is like an EventSource
+    /// (which can also write object), but is intended to log complex objects that can't be serialized.
+    /// </summary>
+    public abstract class TelemetrySource
```

Comment:
We don't expect parameters to be value types (frankly we expect them to be anonymous types most of the time).  The non-allocating property of value types can be achieved by reusing the object being passed, if that is important (high volume always-on code site).     Our experience with EventSource and Project N has given us an appreciation of the downsides of generics, and in this case, it really does look like the value of keeping the precise time is low (it is very likely that clients of this interface will use reflection to implement loose binding protocol).